File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    # 2개의 for문을 통해 이름을 하나씩 접근 / 만약, 이름이 존재하면 해당하는 추억 점수 인덱스에 접근하여 합 구하기
    return [sum(yearning[name.index(i)] for i in p if i in name) for p in photo]

File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> ret;
    
    map<string, int> mp;
    for (int i=0; i<name.size(); i++) mp[name[i]] = yearning[i];
    
    for (vector<string> people: photo) {
        int sum = 0;
        for (string person: people) sum += mp[person];
        ret.push_back(sum);
    }
    return ret;
}